### ISSUE TITLE
[turborepo] can't specify a cache input to the others package

### ISSUE THREAD

**Issue by ZZZJX0912** [vercel/turbo#3087]
### What version of Turborepo are you using?

1.5.3

### What package manager are you using / does the bug impact?

Yarn v1

### What operating system are you using?

Mac

### Describe the Bug

if i specify a input to another package

```
"generate":{
      "inputs": ["config/*","./packages/codegen/*"],
      "outputs": [
        "src/generated/*"
      ]
    },
```
it doesn't bust cache when `codegen` package any files was change 
fwiw i also tried `[project Name]/packages/codegen/*` or `[project Name]/packages/codegen`

### Expected Behavior

i would have expected the cache to bust automatically because a dependency source controlled file changed

### To Reproduce

bug description shows the setup

**Comment by weyert** [comment]
Try using the latest version 1.6.3

**Comment by ZZZJX0912** [comment]
This function is very necessary

**Comment by ZZZJX0912** [comment]
Or provide some ways of inputs screening

**Comment by anthonyshew** [comment]
Closing as duplicate of #2999. See my reply there.

TL;DR: https://turbo.build/blog/turbo-1-12-0#improved-inputs-ergonomics
